Index: app/src/main/java/model/IDataItemCRUDOperations.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package model;\r\n\r\nimport java.util.List;\r\n\r\npublic interface IDataItemCRUDOperations {\r\n    public ToDo createDataItem(ToDo toDo);\r\n\r\n    public List<ToDo> readAllDataItems();\r\n\r\n    public ToDo readDataItem(long id);\r\n\r\n    public ToDo updateDataItem(ToDo toDo);\r\n\r\n    public boolean deleteDataItem(long id);\r\n\r\n    public boolean deleteAllDataItems(boolean remote);\r\n\r\n}\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/model/IDataItemCRUDOperations.java b/app/src/main/java/model/IDataItemCRUDOperations.java
--- a/app/src/main/java/model/IDataItemCRUDOperations.java	
+++ b/app/src/main/java/model/IDataItemCRUDOperations.java	
@@ -2,17 +2,17 @@
 
 import java.util.List;
 
-public interface IDataItemCRUDOperations {
-    public ToDo createDataItem(ToDo toDo);
+public interface IDataItemCRUDOperations
+{
+    ToDo createDataItem(ToDo toDo);
 
-    public List<ToDo> readAllDataItems();
+    List<ToDo> readAllDataItems();
 
-    public ToDo readDataItem(long id);
+    ToDo readDataItem(long id);
 
-    public ToDo updateDataItem(ToDo toDo);
+    boolean updateDataItem(ToDo toDo);
 
-    public boolean deleteDataItem(long id);
+    boolean deleteDataItem(ToDo toDo);
 
-    public boolean deleteAllDataItems(boolean remote);
-
+    boolean authenticateUser(User user);
 }
